packages/api/src/providers/ActivitySendStatusTelemetry/ActivitySendStatusTelemetryComposer.tsx

```diff
@@ -0,0 +1,35 @@
+import { useGetActivityByKey, useSendStatusByActivityKey, useTrackEvent } from '../../hooks';
+
+import usePrevious from '../../hooks/internal/usePrevious';
+
+const ActivitySendStatusTelemetryComposer = () => {
+  const [activityToSendStatusMap] = useSendStatusByActivityKey();
+  const previousActivityToSendStatusMap = usePrevious(activityToSendStatusMap);
+  const getActivityByKey = useGetActivityByKey();
+  const trackEvent = useTrackEvent();
+
+  if (activityToSendStatusMap && previousActivityToSendStatusMap) {
+    const allActivityKeys = activityToSendStatusMap.keys();
+
+    for (const key of allActivityKeys) {
+      const currentStatus = activityToSendStatusMap.get(key);
+      const previousStatus = previousActivityToSendStatusMap.get(key);
+
+      if (!!currentStatus && !!previousStatus && currentStatus !== previousStatus) {

```

Comment:
No need `!!` in `if`, because `if` knows truthy/falsy.

```suggestion
      // `status` is falsy if it is not an outgoing activity.
      // `prevStatus` is falsy if it is not an outgoing activity, or it is recently added.
      // This telemetry data point only emit changes in outgoing activities.
      if (status && prevStatus && status !== prevStatus) {
```

Reply:
In order to track the send status change from "undefined to sending", we need to process the if prevStatus is "undefined". changing the comment:
// `prevStatus` is undefined or a valid status, if it is undefined, it is newly added